Framework/DataObjects/inc/MantidDataObjects/WorkspaceCreation.h

```diff
@@ -158,13 +149,39 @@ std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
       ws = detail::createConcreteHelper<T>();
     }
   }
+  return ws;
+}
 
+MANTID_DATAOBJECTS_DLL void
+initializeFromParent(const API::MatrixWorkspace &parent,
+                     API::MatrixWorkspace &ws);
+}
+
+template <class T, class P, class IndexArg, class HistArg,
+          class = typename std::enable_if<
+              std::is_base_of<API::MatrixWorkspace, P>::value>::type>
+std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
+                          const HistArg &histArg) {
+  std::unique_ptr<T> ws = detail::createUninitialized<T, P>(parent);
   ws->initialize(indexArg, HistogramData::Histogram(histArg));
   detail::initializeFromParent(parent, *ws);
 
   return ws;
 }
 
+template <class T, class P, class IndexArg,
+          class = typename std::enable_if<
+              std::is_base_of<API::MatrixWorkspace, P>::value>::type>
+std::unique_ptr<T> create(const P &parent, const IndexArg &indexArg,
+                          const HistogramData::Histogram &histogram) {
```

Comment:
@soininen Note that it is trivial to do that, since client code can simply use

```cpp
create<EventWorkspace>(parent, BinEdges{1,2,3,4});
```

The histogram with `null` Y and E is created automatically, so there is no burden on the client code.